options/scripting_python.go

```diff
@@ -0,0 +1,100 @@
+package options
+
+import (
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+)
+
+// ScriptingPython defines the configuration of a python environment.
+type ScriptingPython struct {
+	VirtualEnvPath        string            `bson:"virtual_env_path" json:"virtual_env_path" yaml:"virtual_env_path"`
+	RequirementsFilePath  string            `bson:"requirements_path" json:"requirements_path" yaml:"requirements_path"`
+	HostPythonInterpreter string            `bson:"host_python" json:"host_python" yaml:"host_python"`
+	Packages              []string          `bson:"packages" json:"packages" yaml:"packages"`
+	LegacyPython          bool              `bson:"legacy_python" json:"legacy_python" yaml:"legacy_python"`
+	CachedDuration        time.Duration     `bson:"cache_duration" json:"cache_duration" yaml:"cache_duration"`
+	Environment           map[string]string `bson:"env" json:"env" yaml:"env"`
+	Output                Output            `bson:"output" json:"output" yaml:"output"`
+
+	requirementsMTime time.Time
+	cachedAt          time.Time
+	requrementsHash   string
+	cachedHash        string
+}
+
+// NewPythonScriptingEnvironmnet generates a ScriptingEnvironment
+// taking the arguments given for later use. Use this function for
+// simple cases when you do not need or want to set as many aspects of
+// the environment configuration.
+func NewPythonScriptingEnvironmnet(path, reqtxt string, packages ...string) ScriptingEnvironment {
+	return &ScriptingPython{
+		CachedDuration:        time.Hour,
+		HostPythonInterpreter: "python3",
```

Comment:
What about the legacy python case?

Reply:
the constructors are only for default situations. you can always construct directly.